lib/iris/tests/test_netcdf.py

```diff
@@ -964,6 +964,111 @@ def test_no_global_attributes(self):
+            self.assertCDL(filename)
+
+    def test_multiple(self):
+        # Test saving with multiple ancillary variables.
+        testcube = stock.realistic_3d()
+        ancil1_time_lat_lon = iris.coords.AncillaryVariable(
+            np.zeros(testcube.shape), long_name="data_values", units=1
+        )
+        testcube.add_ancillary_variable(ancil1_time_lat_lon, (0, 1, 2))
+        ancil2_time = iris.coords.AncillaryVariable(
+            np.zeros(testcube[:, 0, 0].shape),
+            long_name="time_values",
+            units="s",
+        )
+        testcube.add_ancillary_variable(ancil2_time, 0)
+        ancil3_lon = iris.coords.AncillaryVariable(
+            np.zeros(testcube[0, 0, :].shape),
+            long_name="lon_values",
+            units="m",
+        )
+        testcube.add_ancillary_variable(ancil3_lon, 2)
+        with self.temp_filename(suffix=".nc") as filename:
+            iris.save(testcube, filename)
+            self.assertCDL(filename)
+
+    def test_shared(self):
+        # Check that saving cubes with matching ancillaries maps to a shared
+        # file variable.
+        testcube_1 = stock.realistic_3d()
+        ancil = iris.coords.AncillaryVariable(
+            np.zeros(testcube_1[0].shape), long_name="latlon_refs", units="s"
+        )
+        testcube_1.add_ancillary_variable(ancil, (1, 2))
+
+        testcube_2 = testcube_1.copy()
+        testcube_2.units = "m"
+        testcube_2.rename("alternate_data")
+        # TODO: this should not be necessary, when cube.copy() handles ancils.
+        testcube_2.add_ancillary_variable(ancil.copy(), (1, 2))
+        with self.temp_filename(suffix=".nc") as filename:
+            iris.save([testcube_1, testcube_2], filename)
+            self.assertCDL(filename)
+
+            # Also check that only one, shared ancillary variable was written.
+            ds = nc.Dataset(filename)
+            self.assertIn("air_potential_temperature", ds.variables)
+            self.assertIn("alternate_data", ds.variables)
+            self.assertEqual(
+                ds.variables["air_potential_temperature"].ancillary_variables,
+                "latlon_refs",
+            )
+            self.assertEqual(
+                ds.variables["alternate_data"].ancillary_variables,
+                "latlon_refs",
+            )
+
+    def test_aliases(self):
+        # Check that saving cubes with *differing* ancillaries of the same name
+        # is correctly resolved.
+        testcube_1 = stock.realistic_3d()
+        testcube_2 = testcube_1.copy()
+        testcube_2.units = "m"
+        testcube_2.rename("alternate_data")
+
+        ancil1 = iris.coords.AncillaryVariable(
+            np.zeros(testcube_1[0].shape), long_name="latlon_refs", units="s"
+        )
+        testcube_1.add_ancillary_variable(ancil1, (1, 2))
+
+        ancil2 = ancil1.copy()
+        ancil2.data[0, 0] += 1.0
+        # TODO: needs modifying, when cube.copy() handles ancils.
```

Comment:
As above, this should be possible to address now #3546 is merged.

Reply:
:+1:

Reply:
Note: this actually did *not* need code changing.
The cube is copied before an ancil is attached, so no collision between the two different versions occurs when adding to the second cube.
The two are stored separately (with disambiguated var_names), _because_ they contain different data.  
All good :+1: !